include/rocksdb/table.h

```diff
@@ -27,6 +27,7 @@
 #include "rocksdb/env.h"
 #include "rocksdb/options.h"
 #include "rocksdb/status.h"
+#include "rocksdb/user_defined_index.h"
```

Comment:
Should not need this include if you forward-declare UserDefinedIndexFactory (like the many others below)